docs/python/inference/api_summary.rst

```diff
@@ -133,33 +165,46 @@ Users can bind *OrtValue* (s) directly.
 	io_binding.bind_ortvalue_output('output', Y_ortvalue)
 	session.run_with_iobinding(io_binding)
 
-Device
-======
-
-The package is compiled for a specific device, GPU or CPU.
-The CPU implementation includes optimizations
-such as MKL (Math Kernel Libary). The following function
-indicates the chosen option:
 
-.. autofunction:: onnxruntime.get_device
+You can also bind inputs and outputs directly to a PyTorch tensor.
 
-Examples and datasets
-=====================
-
-The package contains a few models stored in ONNX format
-used in the documentation. These don't need to be downloaded
-as they are installed with the package.
-
-.. autofunction:: onnxruntime.datasets.get_example
+.. code-block:: python
 
-Load and run a model
-====================
+    # X is a PyTorch tensor on device
+    session = onnxruntime.InferenceSession('model.onnx')
```

Comment:
please change to session = onnxruntime.InferenceSession('model.onnx', providers=['CUDAExecutionProvider'])
i think you want to inference on the gpu?
the input and output is on cuda device

We need to reinforce that providers option is required for non default-CPU providers.

Reply:
Yes, thanks. I've been running into this in another PR I am doing. Can you summarize what happens when:
- you don't specify any providers and you are running on CPU (assume this all works?)
- you don't specify any providers and you are running on GPU (assume this errors?)
- you specify CUDA and you are running on CUDA (assume this all works?)

What would happen before this change?

Thanks @jywu-msft!

Reply:
if the ORT build supports CUDA EP (i.e. it was built with CUDA EP enabled), the providers argument to InferenceSession() is required. It will return error if you don't specify providers. (even if you want to run on CPU)
The only time you can create an InferenceSession() without specifying the providers is you are using a CPU only build. in that scenario, the default CPU provider is implied.

Reply:
Great, thank you for explaining that. I will update the docs accordingly

Reply:
Updated